src/write.rs

```diff
@@ -303,14 +318,10 @@ where
         // we need to wait until all epochs have changed since the swaps *or* until a "finished"
         // flag has been observed to be on for two subsequent iterations (there still may be some
         // readers present since we did the previous refresh)
-        //
-        // NOTE: it is safe for us to hold the lock for the entire duration of the swap. we will
-        // only block on pre-existing readers, and they are never waiting to push onto epochs
-        // unless they have finished reading.
-        let epochs = Arc::clone(&self.epochs);
-        let mut epochs = epochs.lock().unwrap();
 
-        self.wait(&mut epochs);
+        let epoch_snapshot = self.epochs.snapshot();
```

Comment:
I think the NOTE that was removed here still applies, it's just about "not seeing new readers" rather than "holding the lock".

Reply:
I tried to update the Note to reflect this now, not quite sure on the wording there though